qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -569,15 +569,21 @@ def _synth_natural_direction(
         if natural_direction in {None, True} and (
             coupling_map or (target is not None and decomposer2q and not preferred_direction)
         ):
-            cmap = coupling_map
-            neighbors0 = cmap.neighbors(qubits[0])
-            zero_one = qubits[1] in neighbors0
-            neighbors1 = cmap.neighbors(qubits[1])
-            one_zero = qubits[0] in neighbors1
-            if zero_one and not one_zero:
-                preferred_direction = [0, 1]
-            if one_zero and not zero_one:
-                preferred_direction = [1, 0]
+            if coupling_map is not None:
+                cmap = coupling_map
+            else:
+                cmap = target.build_coupling_map()
```

Comment:
It is, but it's necessary because of a separate bugfix in `transpile()` that was causing the pass to not be run if you passed in a list of basis gates. See https://github.com/Qiskit/qiskit-terra/pull/7789#issuecomment-1130512064 and the commit message on this initial change: https://github.com/Qiskit/qiskit-terra/pull/7789/commits/eafbf8bfbb74d6033065da9fbb5a5da996e6f734

Reply:
Ok, sounds good.